legacy/src/main/java/org/opensearch/sql/legacy/query/planner/HashJoinQueryPlanRequestBuilder.java

```diff
@@ -59,4 +84,51 @@ public QueryPlanner plan() {
   public Config getConfig() {
     return config;
   }
+
+  /**
+   * Parse JOIN_TIME_OUT(number) hint from SQL string using regex
+   *
+   * @param sql SQL string to parse
+   * @return Optional containing TimeValue if hint found, otherwise empty
+   */
+  private Optional<TimeValue> parseJoinTimeoutFromSql(String sql) {
+    if (sql == null || sql.trim().isEmpty()) {
+      LOG.debug("SQL string is null or empty, no JOIN_TIME_OUT hint to parse");
+      return Optional.empty();
+    }
+
+    try {
+      // Regex pattern to match /*! JOIN_TIME_OUT(number) */
+      Pattern pattern =
+          Pattern.compile(
+              "/\\*!\\s*JOIN_TIME_OUT\\s*\\(\\s*(\\d+)\\s*\\)\\s*\\*/", Pattern.CASE_INSENSITIVE);
+      Matcher matcher = pattern.matcher(sql);
```

Comment:
Using regex is concerning on correctness and performance; can we reuse `HintFactory` / `SqlParser`?